File: leetcode/Easy/moveZeroes.js
var moveZeroes = function(nums) {
    let l = 0
    for (let r = 0; r < nums.length; r++) {
        if (nums[r] !== 0) {
            [nums[l], nums[r]] = [nums[r], nums[l]]
            l++
        }
    }
    return nums
};

console.log(moveZeroes([0,1,0,3,12]))

File: leetcode/Easy/moveZeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = 0
        z = 1

        while z < len(nums):
            if nums[n] == 0:
                if nums[z] != 0:
                    temp = nums[z]
                    nums[z] = nums[n]
                    nums[n] = temp
                    n += 1
                z += 1
            else:
                n += 1
                z += 1
        return nums